test/scenarios.spec.ts

```diff
@@ -221,3 +238,196 @@ suite(
+        const input = TestInput.of(
+            IntentBuilder.of('SetDateEventIntent', {
+                date: '2020-01-01',
+            }),
+        );
+        const result = new ControlResultBuilder(undefined!);
+        await rootControl.canHandle(input);
+        await rootControl.handle(input, result);
+        const dateControlState = findControlById(rootControl, 'dateControl');
+        expect(dateControlState.state.value).eq('2020-01-01');
+    });
+
+    test('Check conflicts in canHandle throws a warn log', async () => {
+        const rootControl = new DateSelectorManager().createControlTree();
+        const input = TestInput.of(
+            SingleValueControlIntent.of(AmazonBuiltInSlotType.DATE, {
+                'AMAZON.DATE': '2018',
+                action: $.Action.Set,
+            }),
+        );
+        const spy = sinon.stub(Logger.prototype, 'warn');
+        const result = new ControlResultBuilder(undefined!);
+        await rootControl.canHandle(input);
+        await rootControl.handle(input, result);
+
+        expect(
+            spy.calledOnceWith(
+                'Custom canHandle function and built-in canHandle function both returned true. Turn on debug logging for more information',
+            ),
+        ).eq(true);
+
+        const dateControlState = findControlById(rootControl, 'dateControl');
+        expect(dateControlState.state.value).eq('2018');
+        expect(result.acts).length(1);
+        expect(result.acts[0]).instanceOf(ValueSetAct);
+    });
+});
+
+suite('== Custom APL Props ==', () => {
+    class ListSelector extends ControlManager {
+        createControlTree(): Control {
+            const topControl = new ContainerControl({ id: 'root' });
+
+            // ListControl
+            const houseControl = new ListControl({
+                id: 'hogwarts',
+                listItemIDs: getCategoriesList(),
+                slotType: 'hogwartsHouse',
+                validation: [
+                    (state, input) =>
+                        getCategoriesList().includes(state.value!)
+                            ? true
+                            : { renderedReason: 'houseControl validation Failed' },
+                ],
+                apl: {
+                    requestValue: {
+                        dataSource: ListControlAPLPropsBuiltIns.APL_TextList_DataSource(
+                            (x) => `Wizard House: ${x}`,
+                        ),
+                        customHandlingFuncs: [
+                            { canHandle: isSelectChoiceEvent, handle: handleSelectChoiceEvent },
+                        ],
+                    },
+                },
+            });
+
+            function getCategoriesList(): string[] {
+                return ['Gryffindor', 'Ravenclaw', 'Slytherin'];
+            }
+
+            function isSelectChoiceEvent(input: ControlInput) {
+                return InputUtil.isIntent(input, 'SelectChoiceEventIntent');
```

Comment:
minor.  for APL custom handling funcs, I was expecting an APL event .. e.g. ButtonPressed event of some kind.  custom Intents would likely go in top-level customHandlerFuncs.

main reason to change to an event is to provide a stronger example for learning.

Reply:
Got it, updated the test cases to handle APL events.